src/Import/src/FMI3/fmi3_import.c

```diff
@@ -53,17 +53,26 @@ fmi3_import_t* fmi3_import_allocate(jm_callbacks* cb) {
     fmu->instanceName = NULL;
     fmu->callbacks = cb;
     fmu->capi = NULL;
-    fmu->md = fmi3_xml_allocate_model_description(cb);
-    jm_vector_init(char)(&fmu->logMessageBufferExpanded,0,cb);
 
+    fmu->md = fmi3_xml_allocate_model_description(cb);
     if (!fmu->md) {
         cb->free(fmu);
         return NULL;
     }
 
+    fmu->termIcon = fmi3_xml_allocate_terminals_and_icons(cb);
+    if (!fmu->termIcon) {
+        fmi3_xml_free_model_description(fmu->md);
+        cb->free(fmu);
+        return NULL;
+    }
+
+    jm_vector_init(char)(&fmu->logMessageBufferExpanded, 0, cb);
```

Comment:
What does this do and/or why is it necessary? Perhaps we can add a comment above to explain it.

Reply:
I think the need to initialize buffers is straight forward, so I added comments to the `fmi3_import_t `struct that should clarify the difference between `logMessageBufferCoded `and `logMessageBufferExpanded`